src/cli/pack_command.zig

```diff
@@ -2123,18 +2164,22 @@ pub const PackCommand = struct {
         return package_json_writer.ctx.writtenWithoutTrailingZero();
     }
 
-    /// A pattern used to ignore or include
-    /// files in the project tree. Might come
-    /// from .npmignore, .gitignore, or `files`
-    /// in package.json
+    /// A glob pattern used to ignore or include files in the project tree.
+    /// Might come from .npmignore, .gitignore, or `files` in package.json
     const Pattern = struct {
         glob: []const u8,
-        /// beginning or middle slash (leading slash was trimmed)
-        rel_path: bool,
-        // can only match directories (had an ending slash, also trimmed)
-        dirs_only: bool,
+        flags: Flags,
 
-        @"leading **/": bool,
+        const Flags = packed struct {
+            /// beginning or middle slash (leading slash was trimmed)
+            rel_path: bool,
+            // can only match directories (had an ending slash, also trimmed)
+            dirs_only: bool,
+
+            @"leading **/": bool,
+            /// true if the pattern starts with `!`
+            negated: bool,
+        };
```

Comment:
This actually won't change the size of Pattern because Pattern remains a normal zig struct. Both versions of this should be 24 bytes (16 for `[]const u8` and 8 for flags + padding)

Reply:
Hmm true if this is 8 byte aligned